### PR TITLE
Weaken tcbSchedDequeue preconditions

### PR DESCRIPTION
**Pull request by ryybrr**
Some lemmas about `tcbSchedDequeue` had preconditions involving `tcb_at'` that were too strong due to the use of `threadGet_const_tcb_at` instead of `threadGet_wp`.

### CODE REVIEW

File: proof/refine/ARM/Schedule_R.thy [hunk 1]
```diff
@@ -319,13 +318,13 @@ lemma setQueue_ksReadyQueues_lift:
   by (wp, clarsimp simp: fun_upd_def snd_def)
 
 lemma tcbSchedDequeue_valid_queues'[wp]:
-  "\<lbrace>valid_queues' and tcb_at' t\<rbrace>
+  "\<lbrace>valid_queues'\<rbrace>
     tcbSchedDequeue t \<lbrace>\<lambda>_. valid_queues'\<rbrace>"
```
Comment (corlewis) [inline proof/refine/ARM/Schedule_R.thy:322]: Very low priority, but I'm a fan of fixing up the indentation and layout of lemmas when I'm already touching them. Quite a few of the lemmas in this PR were previously wrong.

**Review by corlewis — APPROVED** [review]
Nice!

**Review by mktnk3 — APPROVED** [review]
Very nice! This can be merged next when you are ready, I think.

Is it correct that this PR doesn’t have the lemma that is sorried in Corey’s current PR? Does that mean there will be another one coming soon?

**Review by mktnk3 — APPROVED** [review]
Looks all good to me! Please go ahead and merge when you can. You should be able to click the “rebase and merge” option.

**Pull request merged**